src/eval_context.rs

```diff
@@ -1891,9 +1891,11 @@ pub fn resolve<'a, 'tcx>(
     } else {
         let item_type = def_ty(tcx, def_id, substs);
         let def = match item_type.sty {
-            ty::TyFnDef(_, _, f) if
-                f.abi() == Abi::RustIntrinsic ||
-                f.abi() == Abi::PlatformIntrinsic =>
+            ty::TyFnDef(..) if {
+                    let f = item_type.fn_sig(tcx);
```

Comment:
This code is directly copied from [librustc_trans](https://github.com/rust-lang/rust/blob/dc2003b7cc1af7d7beee0b1ceec3d811c28cb35e/src/librustc_trans/monomorphize.rs#L168-L172). Should it also be changed there?